src/Errors.ts

```diff
@@ -0,0 +1,46 @@
+interface ErrorInfoValues {
+  message: string;
+  code: number;
+  statusCode: number;
+}
+
+// TODO: export ErrorInfo from ably-js and use that instead.
+export class ErrorInfo extends Error {
+  code: number;
+  statusCode: number;
+
+  constructor({ message, code, statusCode }: ErrorInfoValues) {
+    super(message);
+
+    if (typeof Object.setPrototypeOf !== 'undefined') {
+      Object.setPrototypeOf(this, ErrorInfo.prototype);
+    }
+
+    this.code = code;
+    this.statusCode = statusCode;
+  }
+}
+
+export const ERR_LOCK_REQUEST_EXISTS = new ErrorInfo({
+  message: 'lock request already exists',
+  code: 40050,
+  statusCode: 400,
+});
+
+export const ERR_LOCK_IS_LOCKED = new ErrorInfo({
+  message: 'lock is currently locked',
+  code: 40051,
+  statusCode: 400,
+});
+
+export const ERR_LOCK_INVALIDATED = new ErrorInfo({
+  message: 'lock was invalidated by a concurrent lock request which now holds the lock',
+  code: 40052,
+  statusCode: 400,
+});
+
+export const ERR_LOCK_RELEASED = new ErrorInfo({
+  message: 'lock was released',
+  code: 40053,
```

Comment:
When should we start adding these error codes to ably-common. My concern is if we leave it to the end, we may forget / have codes scattered across SDKs. Should we perhaps consider doing this when we merge things to `main`?

Reply:
I think they should be added alongside the PR yes, which I didn't do here, but I did ticket it as a priority for next sprint: https://ably.atlassian.net/browse/MMB-238.

I think we might want to reserve a range specifically for Spaces, like we'd done for [Asset Tracking](https://github.com/ably/ably-asset-tracking-common/tree/main/specification#error-codes).